functional/keylime_tenant-signature-verification-options/test.sh

```diff
@@ -0,0 +1,120 @@
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Fail adding of keylime agent and validate IMA file signature verification key which is sign by GPG signature"
+        rlRun "gpg --verify signature-gpg-fake.sig allowlist.txt"
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --sign_verification_key ${limeIMAPublicKey} --signature-verification-key-sig signature-gpg-fake.sig --signature-verification-key-sig-key gpg-key-fake.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartTest "Add keylime agent and validate remote IMA file signature verification key which is sign by GPG signature"
+        rlRun "cp ${limeIMAPublicKey} ."
+        rlRun "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --signature-verification-key-url 'http://localhost:8000/x509_evm.pem' --signature-verification-key-sig-url 'http://localhost:8000/signature-gpg-genuine.sig' --signature-verification-key-sig-url-key gpg-key-genuine.pub -c update"
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Fail adding of keylime agent and validate remote IMA file signature verification key which is sign by GPG signature"
+        rlRun "cp ${limeIMAPublicKey} ."
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --signature-verification-key-url 'http://localhost:8000/x509_evm.pem' --signature-verification-key-sig-url 'http://localhost:8000/signature-gpg-fake.sig' --signature-verification-key-sig-url-key gpg-key-fake.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartCleanup "Do the keylime cleanup"
+        rlRun "kill $HTTP_PID"
+        rlRun "limeStopAgent"
+        rlRun "limeStopRegistrar"
+        rlRun "limeStopVerifier"
+        limeLogfileSubmit $(limeVerifierLogfile)
+        limeLogfileSubmit $(limeRegistrarLogfile)
+        limeLogfileSubmit $(limeAgentLogfile)
+        if limeTPMEmulated; then
+            rlRun "limeStopIMAEmulator"
+            limeLogfileSubmit $(limeIMAEmulatorLogfile)
+            rlRun "limeStopTPMEmulator"
+            rlServiceRestore tpm2-abrmd
+        fi
+        limeClearData
+        limeRestoreConfig
+        limeExtendNextExcludelist ${TESTDIR}
+        rlRun "gpgconf --kill gpg-agent"
```

Comment:
I am not sure but I have a feeling that we should also cleanup oru GPG keys, I guess it could be conflicting with other tests eventually.

Reply:
DONE, removed that dir.

Reply:
That's not what I mean. When you create GPG key I belive it is stored somewhere in `~/.gpg` or similar dir. We should backup this directory in setup phase and restore it later.